packages/fast-form-generator-react/src/form/form-item.array.tsx

```diff
@@ -98,6 +121,28 @@ class FormItemArray extends React.Component<IFormItemArrayProps & IManagedClasse
         );
     }
 
+    public componentDidMount(): void {
+        document.addEventListener("click", this.handleWindowClick);
+    }
+
+    public componentWillUnmount(): void {
+        document.removeEventListener("click", this.handleWindowClick);
+    }
+
+    private handleWindowClick = (e: MouseEvent): void => {
+        if (e.target instanceof Element) {
+            if (!this.optionMenuRef.current.contains(e.target)
```

Comment:
@nalogiudice can you verify why this needs nested ifs?

Reply:
Without the nested ifs, I get this error:
<img width="963" alt="screen shot 2018-08-01 at 11 01 23 am" src="https://user-images.githubusercontent.com/22480660/43539623-42ba0cac-957a-11e8-9add-bd5323973c15.png">

Reply:
I think the casting is wrong, you have `MouseEvent` but this is react, so it should be `React.MouseEvent<HTMLElement>`

Reply:
Looks like since its on the document and not a react component you can use MouseEvent, this shouldn't prevent you from combining the if statements though.